apps/scan/backend/src/app.ts

```diff
@@ -89,6 +89,25 @@ function buildApi(
       const [usbDrive] = await usb.getUsbDrives();
       assert(usbDrive?.mountPoint !== undefined, 'No USB drive mounted');
 
+      const authStatus = await auth.getAuthStatus(
+        constructAuthMachineState(workspace)
+      );
+      if (authStatus.status !== 'logged_in') {
+        await logger.log(
+          LogEventId.BallotPackageConfigAttemptedBeforeAuth,
+          'system'
+        );
+        return err('auth_required_before_ballot_package_load');
+      }
+
+      if (authStatus.user.role !== 'election_manager') {
+        await logger.log(
+          LogEventId.BallotPackageConfigAttemptedByNonElectionManager,
+          authStatus.user.role
+        );
+        return err('user_role_not_allowed');
```

Comment:
Per discussion in https://github.com/votingworks/vxsuite/pull/3147#discussion_r1139246177 I'm ok if we want to throw here instead. Just felt kind of aggressive

Reply:
Yeah the question I always ask here is: Is it possible, even despite code checks, to end up in one of these states? For auth status, there's the complication of external card actions, e.g. removing a card right after triggering an action, making invalid states possible despite code checks. So `err` `Result` objects feel safer than throwing.

I do eventually want to set up auth decorators to abstract away these auth checks. And bake handling of them into the Grout client.